airflow/datasets/__init__.py

```diff
@@ -106,8 +112,90 @@ class DatasetAny(_DatasetBooleanCondition):
 
     agg_func = any
 
+    def __init__(self, *objects: Dataset | DatasetAny | DatasetAll) -> None:
+        """Initialize with one or more Dataset, DatasetAny, or DatasetAll instances."""
+        super().__init__(*objects)
+
+    def __or__(self, other):
+        if isinstance(other, (Dataset, DatasetAny, DatasetAll)):
+            return DatasetAny(*self.objects, other)
+        return NotImplemented
+
+    def __and__(self, other):
```

Comment:
```suggestion
    def __and__(self, other: Dataset | DatasetAny | DatasetAll) -> DatasetAll:
```

Reply:
Best to use `BaseDatasetEventInput` here (and in `__and__` etc)

Reply:
Changed it to `BaseDatasetEventInput`